### ISSUE TITLE
I3042 - `!Sub arn:aws:iam::${AWS::AccountId}:root` identified as having a hardcoded partition or wrongly placed pseudo param

### ISSUE THREAD

**Issue by timzuiddam** [aws-cloudformation/cfn-python-lint#1990]
cfn-lint version: 0.48.3 (latest at the time of writing)

Both `!Sub arn:aws:iam::${AWS::AccountId}:root` and `arn:aws:iam::111122223333:root` identified as having a hardcoded partition or wrongly placed pseudo param.

I haven't checked the code, but as it raises in both scenarios, I reckon the logic regards 'root' as hardcoded while I think this is something that should be an exception to this rule.
You could of course argue that `arn:aws:iam::111122223333:root` has a hardcoded partition and although this is actually used as example everywhere in the CFN documentation, I would be fine if this still raises I3042.

Resource example:
```yaml
  KmsKey:
    Type: AWS::KMS::Key
    Properties:
      Description: Key to encrypt CodeArtifact items
      Enabled: true
      EnableKeyRotation: true
      KeyPolicy:
        Version: 2012-10-17
        Statement:
          - Sid: Allow access for Key Administrators
            Effect: Allow
            Principal:
              AWS:
                - !Sub arn:aws:iam::${AWS::AccountId}:root  # Keep it simple right now
            Action:
              - kms:Create*
              - kms:Describe*
              - kms:Enable*
              - kms:List*
              - kms:Put*
              - kms:Update*
              - kms:Revoke*
              - kms:Disable*
              - kms:Get*
              - kms:Delete*
              - kms:TagResource
              - kms:UntagResource
              - kms:ScheduleKeyDeletion
              - kms:CancelKeyDeletion
            Resource: '*'
          - Sid: Allow direct access to key metadata to the account
            Effect: Allow
            Principal:
              AWS:
                - !Sub arn:aws:iam::${AWS::AccountId}:root  # Keep it simple right now
            Action:
              - kms:Describe*
              - kms:Get*
              - kms:List*
            Resource: '*'
          - Sid: Allow Encrypt/Decrypt
            Effect: Allow
            Principal:
              AWS:
                - !Sub arn:aws:iam::${AWS::AccountId}:root  # Keep it simple right now
            Action:
              - kms:Encrypt
              - kms:Decrypt
              - kms:ReEncrypt*
              - kms:GenerateDataKey*
            Resource: '*'
```

**Comment by PatMyron** [comment]
> I reckon the logic regards 'root' as hardcoded

That rule isn't suggesting changing `root` but rather suggesting [`!Sub arn:${AWS::Partition}:iam::${AWS::AccountId}:root`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/pseudo-parameter-reference.html)

The reasoning for the suggestion is to increase portability of templates across [partitions](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html#arns-syntax)

**Comment by timzuiddam** [comment]
> > I reckon the logic regards 'root' as hardcoded
> 
> That rule isn't suggesting changing `root` but rather suggesting [`!Sub arn:${AWS::Partition}:iam::${AWS::AccountId}:root`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/pseudo-parameter-reference.html)
> 
> The reasoning for the suggestion is to increase portability of templates across [partitions](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html#arns-syntax)

Ah, that partition.. Thanks.
